client/src/app/users/user-list.component.spec.ts

```diff
@@ -39,11 +39,39 @@ const COMMON_IMPORTS: any[] = [
   RouterTestingModule,
 ];
 
-describe('UserListComponent', () => {
+// The `UserListComponent` being tested
+let userList: UserListComponent;
+
+// This constructs the `userList` (declared
+// above) that will be used throughout the tests.
+// This is called in a `beforeEach()` in each of the
+// `describe()` sections below.
+const setupUserList = () =>
+  // Compile all the components in the test bed
+  // so that everything's "ready to go".
+  TestBed.compileComponents().then(() => {
+    // Create a "fixture" of the UserListComponent. that
+    // allows us to get an instance of the component
+    // (userList, below) that we can "control" in
+    // the tests.
+    const fixture = TestBed.createComponent(UserListComponent);
+    userList = fixture.componentInstance;
+    // Tells Angular to "sync" the data bindings between
+    // the model and the DOM. This ensures, e.g., that the
+    // `userList` component actually requests the list
+    // of users from the `MockUserService` so that it's
+    // "up to date" before we start running tests on it.
+    // fixture.detectChanges();
+    fixture.detectChanges();
+  });
```

Comment:
A few suggestions here:

1. Small detail: "set up" as a verb is two words 
2. Prefer `function foo() {` to  `const foo = () =>`
3. Some of the scare-quotes here are likely unnecessary 🙂
4. The comment on line 64 seems left-over from an earlier revision.
5. Small capitalization issue on line 53

Reply:
In this case an async function might be simplest:

```suggestion
async function setUpUserList() {
  // Compile all the components in the test bed
  // so that everything's "ready to go".
  await TestBed.compileComponents();
  // Create a "fixture" of the UserListComponent. That
  // allows us to get an instance of the component
  // (userList, below) that we can control in
  // the tests.
  const fixture = TestBed.createComponent(UserListComponent);
  userList = fixture.componentInstance;
  // Tells Angular to "sync" the data bindings between
  // the model and the DOM. This ensures, e.g., that the
  // `userList` component actually requests the list
  // of users from the `MockUserService` so that it's
  // up to date before we start running tests on it.
  fixture.detectChanges();
}
```

Reply:
But if that's too weird feel free to use the ".then()" version instead 🙂

```suggestion
function setupUserList(): Promise<void> {
  // Compile all the components in the test bed
  // so that everything's "ready to go".
  return TestBed.compileComponents().then(() => {
    // Create a "fixture" of the UserListComponent. That
    // allows us to get an instance of the component
    // (userList, below) that we can control in
    // the tests.
    const fixture = TestBed.createComponent(UserListComponent);
    userList = fixture.componentInstance;
    // Tells Angular to "sync" the data bindings between
    // the model and the DOM. This ensures, e.g., that the
    // `userList` component actually requests the list
    // of users from the `MockUserService` so that it's
    // up to date before we start running tests on it.
    fixture.detectChanges();
  });
}
```

Reply:
I really like the first of those two suggestions and am going to go with it. I'm probably going to change `setUpUserList` to `constructUserList` as I think that's more readable. And frankly I think _all_ the scare quotes are unnecessary. I definitely need a nap. Thanks for all the suggestions!